WebConfTech/website-2019
# Checkout Page Styles

## What does this PR do?

Yes, it adds the styles for #15, but...

### 1: Changes on the `Button`

(I finally added `cursor:pointer` :D)

I added a `:disabled` status , but we'll probably need to wait for @joelalejandro  as what we thought it was `:disabled` it was actually the secondary type.

Also, based on the design spec, I changed the `font-weight` to `lighter`.

### 2: Changes on the `Input`

I added the following _modifiers props_:

1. `bold`: To make the input a little bigger and use `font-weight: bold`. The input size is related to the `font-weight` as it now requires a little more _"breathing space"_ (<- this is just to justify the size change on the `bold` modifier).
2. `borderless`: To remove the input borders.
3. `uppercase`: To make the input text uppercase.

And there's of course **THE HACK** for the webkit/blink autocomplete.

### 3: Changes on `ValidationError`

I renamed the folder `Validation` in order to add the `ValidationMessage` style definition.

### 4: Changes to the `theme`

This is something that will also change once the `Button:disabled` is defined, because I assumed the color was the `:disabled` version of the `secondary` (`$jakarta`) and defined that color as `secondaryDisabled`

> @joelalejandro Even when the issue is solved, how should I call it? `tertiary`?

### 5: Reason why @asermax will hate me

For the mobile view of the form I needed the validation message inside the form, so... I added it as a `children`; not sure if you preferred it as a render prop, feel free to change it.

### 6: Reason why everyone will hate me

The success validation message highlights some words on the desktop view using `em` to give them a different color; but on mobile, it uses quotes and the same style, so... on mobile, the `em::brefore` and `em::after` are `content: '"'`.

## How should it be tested manually?

Go to `/checkout` and compare everything with [the designs](https://xd.adobe.com/spec/f4fda17e-eee3-4d62-4c0e-66e1b7f07aec-82cc/):

1. Yes, the payment for is not ready yet.
2. Make sure to check both desktop and mobile.
